weave/weave_client.py

```diff
@@ -142,6 +144,8 @@ def map_to_refs(obj: Any) -> Any:
         return [map_to_refs(v) for v in obj]
     elif isinstance(obj, dict):
         return {k: map_to_refs(v) for k, v in obj.items()}
+    elif isinstance(obj, WeaveObject):
+        return map_to_refs(obj._val)
```

Comment:
this branch happens when we save a dirty object that is then stripped of the ref, correct? I wonder if this implies that stripping the ref should return the inner `val._val`?

Reply:
Do you mean in `_save_object`?  I've done it this way because some objects don't have `val._val`, e.g. `WeaveList`, or `WeaveDict`